awswrangler/s3/_write_dataset.py

```diff
@@ -338,18 +338,24 @@ def _to_buckets_distributed(  # pylint: disable=unused-argument
 ) -> List[str]:
     df_groups = df.groupby(by=_get_bucketing_series(df=df, bucketing_info=bucketing_info))
     paths: List[str] = []
+
     df_paths = df_groups.apply(
         func.dispatch(ModinDataFrame),  # type: ignore
         path_root=path_root,
         filename_prefix=filename_prefix,
-        boto3_session=None,
+        boto3_session=_utils.boto3_to_primitives(boto3_session=boto3_session),
```

Comment:
I think (before you joined) it had been agreed that in the distributed case the boto3_session would not be passed. I see you also do it for the text [read_datasource](https://github.com/aws/aws-sdk-pandas/blob/release-3.0.0/awswrangler/s3/_read_text.py#L158) while we don't do it for the parquet read_datasource or the `_to_partitions_distributed` above.

Happy to revisit this but would want it to be consistent across the codebase regardless of the decision. FYI @kukushking

Reply:
My logic for including was that if the customer specified a region that's different from the default region, or a profile different from the different profile when initializing that session, they'd expect our code to follow that preference if they pass the session to our functions.

Anyway, these are just the properties behind the boto3 session, no? Or is it passing information other than the [parameters](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html) needed to initialize a session?

Reply:
The main issue we had with boto3_session is that it's not pickable and therefore it cannot be passed as is and must be broken down into primitives. We assumed that users of the distributed version would just use whatever session is in the EC2 or Manta environment but I agree that it's restrictive. If we wish to roll this back however, we would need to do it in other parts like [here](https://github.com/aws/aws-sdk-pandas/blob/release-3.0.0/awswrangler/distributed/_pool.py#L19) where we currently discard the boto3 session altogether

Reply:
I didn't think about the fact that they'd be distributing this to different EC2 instances which will have their own roles. You're right, I'll remove this.